x-pack/plugins/observability_ai_assistant/public/components/prompt_editor/prompt_editor_function.tsx

```diff
@@ -72,6 +80,12 @@ export function ChatPromptEditorFunction({ functionName, functionPayload, onChan
     }
   }, [functionName, functionPayload, initialJsonString, onChange, previousPayload]);
 
+  useEffect(() => {
+    return () => {
+      onBlur();
+    };
+  }, [onBlur]);
```

Comment:
Hmm this looks a little dangerous in terms of consumers passing a new callback every render?

Reply:
Its in the return of the useEffect though, so it should only run on unmount, no?

Reply:
oh right, it just re-creates the unmount hook. I think it's better to use a ref here though, that way we don't call this hook on every render. I'm not too worried about performance, but more so about bugs creeping in and testability.

Reply:
That was my first thought as well, unfortunately Monaco (or at least Kibana's implementation of it) does not have an `onBlur`. 

FWIW it also does not have an `onFocus`. It only has an `editorDidMount` prop, which can be used as an alternative for `onFocus`.

Reply:
sorry, I meant `onBlurRef = useRef(props.onBlur); onBlurRef.current = props.onBlur;`

Reply:
Turns out the monaco editor instance has a `onDidBlurEditorWidget` event. 

Implemented with:

```
  editorRef.current?.onDidBlurEditorWidget(() => {
    onBlur();
  });
```